app/controllers/application_controller.rb

```diff
@@ -1,2 +1,21 @@
+# frozen_string_literal: true
+
 class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+  include SessionsHelper
+
+  def log_in(user)
+    session[:user_id] = user.id
+    user.remember
+    user.save
+    cookies.permanent[:remember_token] = user.remember_token
+
+    redirect_to(session[:forwarding_url] || user)
+  end
+
+  def sing_out
```

Comment:
Sing Out? Really? I mean, if you want to name it like that, by all means, go ahead, but I initially thought this was a typo or a mistake on your part; thanks for the :rofl: